packages/dashboard-core-plugins/src/linker/LinkerOverlayContent.tsx

```diff
@@ -64,23 +71,41 @@ export class LinkerOverlayContent extends Component<
 
     this.handleBlur = this.handleBlur.bind(this);
     this.handleMouseMove = this.handleMouseMove.bind(this);
+    this.handleMouseDown = this.handleMouseDown.bind(this);
+    this.handleMouseUp = this.handleMouseUp.bind(this);
     this.handleKeyDown = this.handleKeyDown.bind(this);
     this.handleKeyUp = this.handleKeyUp.bind(this);
     this.handleEscapePressed = this.handleEscapePressed.bind(this);
     this.handleOperatorChanged = this.handleOperatorChanged.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+
+    this.dialogRef = React.createRef();
 
     this.state = {
       mouseX: undefined,
       mouseY: undefined,
+      dialogX: undefined,
+      dialogY: undefined,
+      offsetX: 0,
+      offsetY: 0,
+      isDragging: false,
       mode: 'select',
     };
   }
 
   componentDidMount(): void {
     window.addEventListener('blur', this.handleBlur, true);
     window.addEventListener('mousemove', this.handleMouseMove, true);
+    window.addEventListener('mouseup', this.handleMouseUp, true);
     window.addEventListener('keydown', this.handleKeyDown, true);
     window.addEventListener('keyup', this.handleKeyUp, true);
+    window.addEventListener('resize', this.handleResize, true);
+    this.setState({
+      dialogX:
+        window.innerWidth - (this.dialogRef.current?.offsetWidth ?? 0) - 16,
```

Comment:
Where'd the `16` come from? Extract it into a constant at the top of the file and name is something descriptive, e.g.
```
const DIALOG_MARGIN = 16;
```